upgrade_test.py

```diff
@@ -23,6 +22,36 @@
 from sdcm.nemesis import RollbackNemesis
 from sdcm.nemesis import UpgradeNemesis
 from sdcm.fill_db_data import FillDatabaseData
+from functools import wraps
+
+
+def truncate_entries(func):
+    @wraps(func)
+    def _inner(self, *args, **kwargs):
+        # Perform validation of truncate entries in case the new version is 3.1 or more
+        node = args[0]
+        new_version = self.params.get('new_version', default='')
+        new_ver = '.'.join(n for n in new_version.split('.')[:2])
+        if new_ver and float(new_ver) >= 3.1:
+            base_version = self.params.get('scylla_version', default='')
+            system_truncated = True if base_version.startswith('3.1') else False
+
+            with self.cql_connection_patient(node) as session:
+                self.cql_truncate_table_for_truncate_test(session=session)
+                self.validate_truncated_entries_for_table(session=session, system_truncated=system_truncated)
+
+        func(self, *args, **kwargs)
+
+        result = node.remoter.run('scylla --version')
+        new_ver = result.stdout
+        new_ver = '.'.join(n for n in new_ver.split('.')[:2])
+        if new_ver and float(new_ver) >= 3.1:
+            # re-new connection
+            with self.cql_connection_patient(node) as session:
```

Comment:
better: self.cql_connection_patient(node, keyspace="truncate_ks") and remove all `USE truncate_ks` in the helper functions

Reply:
Fixed